blog/2023-02-12-github-mysql/index.md

```diff
@@ -0,0 +1,356 @@
+---
+title: Build a notification system that pushes any Github event to Slack in 5 minutes
+authors: Lucas
+tags: ['GitHub', 'Message Queue', 'MySQL', 'data analysis']
+---
+
+## How to store and analyze GitHub data in MySQL
+
+If you are running an open-source project, I believe you must be interested in the following questions: 
+
+* Has your project received more and more attention recently? 
+
+* What organizations are having developers leaving your project at an accelerated rate？
+
+* Which organizations of developers are being attracted to your project?
+
+* What strategy should be adopted to attract more contributors?
+
+Analyzing GitHub data can not only answer the above questions. And it can help you gain insight into important trends in more open-source projects. For example, it can help open source operators gain real-time insight into the trends of project developers, the latest trends of contributors, which organizations pay attention to your project, and so on.
+
+This blog will use Vanus to build a data pipeline from GitHub to MySQL, so as to help developers store GitHub data in MySQL in real time. At the same time, some examples are given to help developers analyze GitHub data to gain insight into trends. The results are shown below:
+
+![githubdata-mysql](./img/githubdata-mysql.jpeg)
+
+
+
+## Table of Contents
+
+- [What is GitHub](#What-is-GitHub)
+  - [About GitHub](#About-GitHub)
+  - [What are GitHub events](#What-are-GitHub-events)
+- [What is MySQL](#What-is-MySQL)
+- [How to Connect GitHub to MySQL](#How-to-Connect-GitHub-to-MySQL)
+  - [Prerequisites](#Prerequisites)
+  - [Step 1: Deploying Vanus in the playground](#Step-1-Deploying-Vanus-in-the-playground)
+  - [Step 2: Deploy the GitHub source connector](#Step-2-Deploy-the-GitHub-source-connector)
+  - [Step 3:Deploy MySQL on Docker](#Step-3-Deploy-MySQL-on-Docker)
+  - [Step 4: Deploy the MySQL sink connector](#Step-4-Deploy-the-MySQL-sink-connector )
+- [Store Github star events to MySQL](#Store-Github-star-events-to-MySQL)
+- [Analyze GitHub data](#Analyze-GitHub-data)
+- [Conclusion](#Conclusion)
+
+## What is GitHub
+### About GitHub
+
+GitHub is an online software development platform. It's used for storing, tracking, and collaborating on software projects. It makes it easy for developers to share code files and collaborate with fellow developers on open-source projects. GitHub also serves as a social networking site where developers can openly network, collaborate, and pitch their work.
+
+Since its founding in 2008, GitHub has acquired millions of users and established itself as a go-to platform for collaborative software projects. This free service comes with several helpful features for sharing code and working with others in real-time
```

Comment:
```suggestion
Since its founding in 2008, GitHub has acquired millions of users and established itself as a go-to platform for collaborative software projects. This free service has several helpful features for sharing code and working with others in real time.
```